src/Fetch_Abort.res

```diff
@@ -1,10 +1,39 @@
+module AbortEvent = {
+  type t
+}
+
 module Signal = {
   type t
 
+  /**
+  https://dom.spec.whatwg.org/#dom-abortsignal-any
+  this is new as of 2024
+   */
+  @scope("AbortSignal")
+  external any: array<t> => t = "any"
+
+  @scope("AbortSignal")
+  external timeout: int => t = "timeout"
+
+  @scope("AbortSignal")
+  external abort: 'reason => t = "abort"
```

Comment:
`reason` is optional here. With uncurried functions we can now model that with an optional argument:
```suggestion
  @scope("AbortSignal")
  external abort: (~reason: 'reason=?) => t = "abort"
```

Reply:
And could you make it an optional argument in `Controller.abort` as well?

Reply:
Yeah, makes sense. This is/was an attempt to avoid needing to explicitly pass `~reason` in v10, but I suppose you'd either have to do `Fetch.Signal.abort(signal, None)` or just make it slightly more verbose with the labelled argument regardless